src/utils/MainApi.js

```diff
@@ -14,11 +14,11 @@ export default class Auth {
     return Promise.reject(`Error:${res.status}`); //returning Error status
   }
 
-  register({ email, password, username }) {
+  register({ email, password, name }) {
     return fetch(`${this.baseUrl}/signup`, {
       method: "POST",
       headers: this.headers,
-      body: JSON.stringify({ email, password, username }),
+      body: JSON.stringify({ email, password, name }),
     }).then(this.checkResponse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.